lib/whitespace.coffee

```diff
@@ -31,14 +31,26 @@ class Whitespace
         if atom.config.get('whitespace.ensureSingleTrailingNewline', scope: scopeDescriptor)
           @ensureSingleTrailingNewline(editor)
 
+    editorTextInsertedSubscription = editor.onDidInsertText (event) =>
+      return unless event.text is '\n'
+      scopeDescriptor = editor.getRootScopeDescriptor()
+
+      if atom.config.get('whitespace.removeTrailingWhitespace', scope: scopeDescriptor)
+        unless atom.config.get('whitespace.ignoreWhitespaceOnlyLines', scope: scopeDescriptor)
+          editor.transact =>
```

Comment:
There's still a case here where we call `transact` but don't do anything in the transaction: when the previous row is not blank.

You might not even need transact at all actually; is there a reason you wrapped this in a `transact`?